judge/views/contests.py

```diff
@@ -380,13 +387,9 @@ def join_contest(self, request, access_code=None):
                     break
         else:
             SPECTATE = ContestParticipation.SPECTATE
```

Comment:
Nit: Delete this, and replace `virtual=SPECTATE` with `virtual=ContestParticipation.SPECTATE`.